extension/data_loader/file_data_loader.cpp

```diff
@@ -189,7 +191,12 @@ Result<FreeableBuffer> FileDataLoader::load(
   size_t needed = size;
   uint8_t* buf = reinterpret_cast<uint8_t*>(aligned_buffer);
   while (needed > 0) {
-    ssize_t nread = ::read(fd_, buf, needed);
+    // read on macos would fail with EINVAL if size > INT32_MAX
```

Comment:
Please begin comments with uppercase letters and end with a period, to be consistent with the rest of the file. Thanks for adding this comment, though, to explain why this extra logic is necessary, and where to focus testing if it changes.

```suggestion
    // Reads on macOS will fail with EINVAL if size > INT32_MAX.
```

Reply:
When you get a chance please update this comment, then we can land the PR

Reply:
Done.